airflow/providers/databricks/hooks/databricks.py

```diff
@@ -156,13 +156,13 @@ def submit_run(self, json: dict) -> int:
         return response["run_id"]
 
     def list_jobs(
-        self, limit: int = 25, offset: int = 0, expand_tasks: bool = False, job_name: str | None = None
+        self, limit: int = 25, page_token: str | None = None, expand_tasks: bool = False, job_name: str | None = None
```

Comment:
this is a breaking change. We should deprecate first.

Reply:
Are people supposed to use hook directly, especially for such advanced cases like offsets?

Reply:
It has been deprecated in [Databricks API docs](https://docs.databricks.com/api/workspace/jobs/list) for a few months now if that helps

Reply:
Also, Databricks has already gone through the breaking change process to communicate to the customers who are using the JobsList API with offsets (through Airflow), urging them to update ASAP @eladkal

Reply:
If both `offset` and `page_token` are supported by the API then we need to support them both with raising deprecation warning when using `offset`

Reply:
@eladkal Done, enabled both with a deprecation warning